keps/sig-node/2371-cri-pod-container-stats/README.md

```diff
@@ -272,6 +273,28 @@ Thus, this KEP largely the plan described [here](#plan), with some changes:
 
 #### Open Questions
 1. For the newly introduced CRI API fields, should there be Windows and Linux specific fields?
+    For the alpha implementation, [PR #102789](https://github.com/kubernetes/kubernetes/pull/102789) added a Linux specific field `LinuxPodSandboxStats`. It also added Windows Specific field called  `WindowsPodSandboxStats` but left it blank.  
+
+    The proposal for Widnows support is to add a fields to `PodSandboxStats` which will be used by both Linux and Windows.  Windows and
+    Linux would both use this field filling in stats that make most sense for them.  Windows is currently missing several stats but can reasonably fill in many of the missing fields (see the [Windows](#windows) section below).  This would be similar approach that is use today for the current CRI `ListContainerStats` call and this would make logic in kubelet more straight forward.
+
+    The field `LinuxPodSandboxStats` would stay the same (with duplicated fields) for backwards compatibility.  We will also leave `WindowsPodSandboxStats` with a comment that it is not currently used.   This option allows us to add specific fields in the future, Linux and Windows sub fields could be added to `LinuxPodSandboxStats` and `WindowsPodSandboxStats` if there are use cases for specific fields.  
+
+    The downside to this approach is that we have two fields (`LinuxPodSandboxStats` and `PodSandboxStats`) which are initially identical. With this approach, We will mark the `LinuxPodSandboxStats` and `WindowsPodSandboxStats`  structure as depreciated in favor of a single structure.
+
+    Alternatives include:
```

Comment:
Did you consider an unstructured approach? I.e. let the CRI implementation decide what stats to return, along with definitions (probably a prometheus-like format)?

To maintain backwards compatibility with the existing summary & cadvisor kubelet APIs, there could be a mapping of well-known metric names to fields in those APIs.

Reply:
for metrics reported through the CRI, they'll eventually be translated into kubelet's `/stats/summary` endpoint. Since it's an API with a guaranteed lifecycle, I think converting from unstructured API to a structured could be weird.

I think the utility of having windows or linux specific metrics comes from the possibility of extending that API. Though, frankly, I'm not really sure we'll need those structures, as `/stats/summary` is very stable at this point.

The CRI's implementing `/metrics/cadvisor` (cadvisors prometheus endpoint) is not passed through CRI, and is unstructured. This could be a place where such unstructured, platform-specfic metrics could be reported